saltproc/openmc_depcode.py

```diff
@@ -283,42 +563,70 @@ def write_saltproc_openmc_tallies(self, materials, geometry):
 
-        tally = openmc.Tally(name='total-fission-neutrons')
-        tally.filters = [openmc.UniverseFilter(geometry.root_universe)]
-        tally.scores = ['nu-fission']
-        tallies.append(tally)
+        delayed_kwargs = {'domain': geometry.root_universe,
+                          'domain_type': 'universe',
+                          'energy_groups': energy_groups,
+                          'delayed_groups': delayed_groups}
+
+        delayed_lambda = DecayRate(**delayed_kwargs)
+        beta = Beta(**delayed_kwargs)
 
-        tally = openmc.Tally(name='precursor-decay-constants')
-        tally.filters = [openmc.DelayedGroupFilter([1, 2, 3, 4, 5, 6])]
-        tally.scores = ['decay-rate']
+        self._delayed_lambda = delayed_lambda
+        self._beta = beta
+
+        tallies = openmc.Tallies()
+        tallies += list(delayed_lambda.tallies.values())
+        tallies += list(beta.tallies.values())
+
+        tally = openmc.Tally(name='breeding_ratio_tally')
+        tally.filters = [openmc.UniverseFilter(geometry.root_universe)]
+        tally.scores = ['(n,gamma)', 'absorption']
+        tally.nuclides = ff_nuclides
         tallies.append(tally)
 
-        tally = openmc.Tally(name='fission-energy')
+        tally = openmc.Tally(name='fission_energy')
         tally.filters = [openmc.UniverseFilter(geometry.root_universe)]
-        tally.scores = [
-            'fission-q-recoverable',
-            'fission-q-prompt',
-            'kappa-fission']
+        tally.scores = ['kappa-fission']
         tallies.append(tally)
 
-        tally = openmc.Tally(name='normalization-factor')
+        tally = openmc.Tally(name='heating')
         tally.filters = [openmc.UniverseFilter(geometry.root_universe)]
         tally.scores = ['heating']
         tallies.append(tally)
 
-        out_path = os.path.dirname(self.runtime_inputfile['settings'])
-        self.runtime_inputfile['tallies'] = \
-            os.path.join(out_path, 'tallies.xml')
+        self.runtime_inputfile['tallies'] = self.output_path / 'tallies.xml'
         tallies.export_to_xml(self.runtime_inputfile['tallies'])
-        del tallies
 
+    def _get_fissile_fertile_nuclides(self):
+        nuclides_with_data = set()
+        data_lib = DataLibrary.from_xml()
+        for library in data_lib.libraries:
+            if library['type'] != 'neutron':
+                continue
+            for name in library['materials']:
+                if name not in nuclides_with_data:
+                    nuclides_with_data.add(name)
+
+        chain = Chain.from_xml(self.chain_file_path)
+        burnable_nucs = [nuc.name for nuc in chain.nuclides
+                         if nuc.name in nuclides_with_data]
+
+        self._fissile_nucs = set([nuc for nuc in _FISSILE_NUCLIDES if nuc in burnable_nucs])
+        self._fertile_nucs = set([nuc for nuc in _FERTILE_NUCLIDES if nuc in burnable_nucs])
 
+        return list(self._fissile_nucs.union(self._fertile_nucs))
```

Comment:
Docstrings

Reply:
I personally think the function name is pretty descriptive. It gets the fissile and fertile nuclides.